### PR TITLE
HDDS-1895. Support Key ACL operations for OM HA.

### PR DESCRIPTION
**Pull request by bharatviswa504**
This patch is dependent on HDDS-1884.
Posted this PR on top of HDDS-1884.
2nd commit is the actual changes for this PR.

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 44 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 11 | Maven dependency ordering for branch |
| +1 | mvninstall | 584 | trunk passed |
| +1 | compile | 373 | trunk passed |
| +1 | checkstyle | 77 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 873 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 161 | trunk passed |
| 0 | spotbugs | 413 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 606 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 23 | Maven dependency ordering for patch |
| +1 | mvninstall | 559 | the patch passed |
| +1 | compile | 389 | the patch passed |
| +1 | cc | 389 | the patch passed |
| +1 | javac | 389 | the patch passed |
| -0 | checkstyle | 50 | hadoop-ozone: The patch generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 680 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 159 | the patch passed |
| +1 | findbugs | 633 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 282 | hadoop-hdds in the patch passed. |
| -1 | unit | 1972 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 45 | The patch does not generate ASF License warnings. |
| | | 7739 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.pipeline.TestPipelineClose |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestContainerStateMachine |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.client.rpc.TestContainerStateMachineFailures |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1230 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux c2829efda7ee 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / d6697da |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/1/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/1/testReport/ |
| Max. process+thread count | 4999 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/integration-test U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 43 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 32 | Maven dependency ordering for branch |
| +1 | mvninstall | 590 | trunk passed |
| +1 | compile | 372 | trunk passed |
| +1 | checkstyle | 72 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 811 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 148 | trunk passed |
| 0 | spotbugs | 428 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 619 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 19 | Maven dependency ordering for patch |
| +1 | mvninstall | 576 | the patch passed |
| +1 | compile | 387 | the patch passed |
| +1 | cc | 387 | the patch passed |
| +1 | javac | 387 | the patch passed |
| -0 | checkstyle | 33 | hadoop-ozone: The patch generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 644 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 158 | the patch passed |
| +1 | findbugs | 688 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 198 | hadoop-hdds in the patch failed. |
| -1 | unit | 2772 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 41 | The patch does not generate ASF License warnings. |
| | | 8368 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.ozoneimpl.TestOzoneContainer |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1230 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux dc1375ba6c49 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 3cc0ace |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/2/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/2/testReport/ |
| Max. process+thread count | 3635 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/integration-test U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 54 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 11 | Maven dependency ordering for branch |
| +1 | mvninstall | 618 | trunk passed |
| +1 | compile | 366 | trunk passed |
| +1 | checkstyle | 77 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 843 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 154 | trunk passed |
| 0 | spotbugs | 449 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 653 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 17 | Maven dependency ordering for patch |
| +1 | mvninstall | 555 | the patch passed |
| +1 | compile | 373 | the patch passed |
| +1 | cc | 373 | the patch passed |
| +1 | javac | 373 | the patch passed |
| -0 | checkstyle | 38 | hadoop-ozone: The patch generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 650 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 146 | the patch passed |
| +1 | findbugs | 618 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 302 | hadoop-hdds in the patch failed. |
| -1 | unit | 2210 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 39 | The patch does not generate ASF License warnings. |
| | | 7888 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.block.TestBlockManager |
|   | hadoop.ozone.om.TestKeyManagerImpl |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1230 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 8fdb94c2cae7 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 00b5a27 |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/3/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/3/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/3/testReport/ |
| Max. process+thread count | 4470 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/integration-test U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by bharatviswa504** [comment]
Rebased on top of the latest trunk, as HDDS-1884 got checked in.
Now it is ready for review.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 43 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 33 | Maven dependency ordering for branch |
| +1 | mvninstall | 597 | trunk passed |
| +1 | compile | 373 | trunk passed |
| +1 | checkstyle | 76 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 870 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 164 | trunk passed |
| 0 | spotbugs | 420 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 617 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 25 | Maven dependency ordering for patch |
| +1 | mvninstall | 561 | the patch passed |
| +1 | compile | 374 | the patch passed |
| +1 | javac | 374 | the patch passed |
| +1 | checkstyle | 85 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 690 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 161 | the patch passed |
| +1 | findbugs | 632 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 395 | hadoop-hdds in the patch passed. |
| -1 | unit | 1727 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 44 | The patch does not generate ASF License warnings. |
| | | 7652 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineProvider |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.om.TestKeyManagerImpl |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1230 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux c5e8fb518c6a 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 88ed1e0 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/4/testReport/ |
| Max. process+thread count | 5001 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/integration-test U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/utils/OzoneManagerRatisUtils.java [hunk 1]
```diff
@@ -141,20 +144,26 @@ private static OMClientRequest getOMAclRequest(OMRequest omRequest) {
         return new OMVolumeAddAclRequest(omRequest);
       } else if (ObjectType.BUCKET == type) {
         return new OMBucketAddAclRequest(omRequest);
+      } else if (type == ObjectType.KEY) {
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/utils/OzoneManagerRatisUtils.java]: The code is using reverse order for equality. Can you use the same?

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAddAclRequest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  private static final Logger LOG =
+      LoggerFactory.getLogger(OMKeyAddAclRequest.class);
+
+  private static BooleanBiFunction<List<OzoneAclInfo>, OmKeyInfo> keyAddAclOp;
+  private String path;
+  private List<OzoneAclInfo> ozoneAcls;
+
+  static {
+    keyAddAclOp = (ozoneAcls, omKeyInfo) -> {
+      return omKeyInfo.addAcl(ozoneAcls.get(0));
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAddAclRequest.java]: Sorry I didn't understand this. What is it doing? Felt little odd to see a lambda in a static block.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAddAclRequest.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+  @Override
+  void onComplete(boolean operationResult, IOException exception,
+      OMMetrics omMetrics) {
+    if (operationResult) {
+      LOG.debug("Add acl: {} to path: {} success!", getAcls(), getPath());
+    } else {
+      omMetrics.incNumBucketUpdateFails();
+      if (exception == null) {
+        LOG.error("Add acl {} to path {} failed, because acl already exist",
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAddAclRequest.java]: Should we just make this info or even debug?

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAddAclRequest.java (hunk 2 above)
Comment (bharatviswa504) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAddAclRequest.java]: As discussed offline, have changed it.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/utils/OzoneManagerRatisUtils.java (hunk 1 above)
Comment (bharatviswa504) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/utils/OzoneManagerRatisUtils.java]: Updated it.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAddAclRequest.java (hunk 3 above)
Comment (bharatviswa504) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAddAclRequest.java]: Done.

**Comment by bharatviswa504** [comment]
Thank You @arp7 for the offline discussion.
Addressed the review comments.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 41 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 16 | Maven dependency ordering for branch |
| +1 | mvninstall | 598 | trunk passed |
| +1 | compile | 374 | trunk passed |
| +1 | checkstyle | 74 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 867 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 166 | trunk passed |
| 0 | spotbugs | 423 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 619 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 25 | Maven dependency ordering for patch |
| +1 | mvninstall | 567 | the patch passed |
| +1 | compile | 380 | the patch passed |
| +1 | javac | 380 | the patch passed |
| +1 | checkstyle | 82 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 681 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 172 | the patch passed |
| +1 | findbugs | 641 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 297 | hadoop-hdds in the patch passed. |
| -1 | unit | 1764 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 54 | The patch does not generate ASF License warnings. |
| | | 7608 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineProvider |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.om.TestKeyManagerImpl |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1230 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 11ed57c004ba 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 98dd7c4 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/5/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/5/testReport/ |
| Max. process+thread count | 5365 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/integration-test U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/5/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 82 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 38 | Maven dependency ordering for branch |
| +1 | mvninstall | 624 | trunk passed |
| +1 | compile | 378 | trunk passed |
| +1 | checkstyle | 75 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 963 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 173 | trunk passed |
| 0 | spotbugs | 466 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 684 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 21 | Maven dependency ordering for patch |
| +1 | mvninstall | 583 | the patch passed |
| +1 | compile | 478 | the patch passed |
| +1 | javac | 478 | the patch passed |
| +1 | checkstyle | 92 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 888 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 215 | the patch passed |
| +1 | findbugs | 770 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 421 | hadoop-hdds in the patch passed. |
| -1 | unit | 3537 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 50 | The patch does not generate ASF License warnings. |
| | | 10264 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestReadRetries |
|   | hadoop.ozone.scm.TestSCMNodeManagerMXBean |
|   | hadoop.ozone.scm.TestSCMMXBean |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.hdds.scm.container.TestContainerStateManagerIntegration |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.web.client.TestKeysRatis |
|   | hadoop.ozone.client.rpc.TestKeyInputStream |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1230 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux ab345089517b 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 98dd7c4 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/6/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/6/testReport/ |
| Max. process+thread count | 3462 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/integration-test U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1230/6/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by arp7** [comment]
I committed this. None of the test failures looks related.

**Comment by cawatson** [comment]
I am creating a tool to aid developers by partially automating unit testing.
Could you speak to the validity of these generated assert statement for the given test method?
All test methods come from the file: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHA.java

Test Method: testAllBucketOperations
Assert Statement(s): 
`org . junit . Assert . assertTrue ( versioning . getVersioning ( ) )`
`org . junit . Assert . assertTrue ( OzoneTestUtils . getVersioning ( ) )`
`org . junit . Assert . assertTrue ( OzoneBucket . getVersioning ( ) )`

Test Method: testFileOperationsWithNonRecursive
Assert Statement(s):
`org . junit . Assert . assertNotNull ( OMException )`
`org . junit . Assert . assertNotNull ( keyName )`

Test Method: testMultipartUploadWithOneOmNodeDown
Assert Statement(s):
`org . junit . Assert . assertNotNull ( keyName )`
`org . junit . Assert . assertTrue ( objectStore . getOMProxyProvider ( node ) )`

Test Method: testOMProxyProviderFailoverToCurrentLeader
Assert Statement(s): 
`org . junit . Assert . assertTrue ( ( ( leaderOMNodeId . createVolumeTest ( node ) ) == 1 ) )`
`org . junit . Assert . assertTrue ( ( ( leaderOMNodeId . createVolumeTest ( node ) ) == 0 ) )`
`org . junit . Assert . assertTrue ( ( ( leaderOMNodeId . createVolumeTest ( node ) ) > 0 ) )`

Test Method: testOMRetryProxy
Assert Statement(s): 
`org . junit . Assert . assertTrue ( ( ( Logger . addAppender ( node ) ) == null ) )`

Test Method: testReadRequest
Assert Statement(s): 
`org . junit . Assert . assertEquals ( 0 , currentLeaderNodeId . count ( 10 ) )`

Thank you for your time.

**Pull request merged**